### PR TITLE
Revocation list

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+const vc = require('vc-js');
+const {util: {BedrockError}} = require('bedrock');
+const Collection = require('./Collection');
+const {EdvClient, EdvDocument} = require('edv-client');
+
+// load config defaults
+require('./config');
+
+exports.issue = async ({credential, suite}) => {
+  return await vc.issue({credential, documentLoader, suite});
```
Comment (mattcollier) [inline lib/helpers.js:28]:
```suggestion
  return vc.issue({credential, documentLoader, suite});
```

File: lib/http.js [hunk 2]
```diff
@@ -112,4 +120,81 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
 
       res.json({verifiableCredential});
     }));
+
+  // pull latest RLC from EDV storage for an instance and publish it
+  app.post(
+    routes.publishRlc,
+    // TODO: also accept bearer token to identify profile agent in the future
```
Comment (mattcollier) [inline lib/http.js:127]: looks like this has already been done?

File: lib/CredentialStatusWriter.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+      const {indexAssignmentEdvDoc, item: {rlSequence}} = listShard;
+      const doc = await indexAssignmentEdvDoc.read();
+
+      // 1.2. If the IAD's RL sequence matches the one from the LS and the
+      //   IAD's latest assigned index value is behind the index in the VC's
+      //   `credentialStatus` field:
+      const {revocationListIndex} = credentialStatus;
+      const localIndex = _getLocalIndex({listShard, revocationListIndex});
+      if(doc.content.rlSequence === rlSequence &&
+        doc.content.nextLocalIndex < localIndex) {
```
[the remark this answers is not in the document]
Comment (dlongley) [reply lib/CredentialStatusWriter.js:142]: No, if this conditional is met it just means that another process either failed to update the IAD or is concurrently using it -- and we're cleaning up/taking over. If the conditional is not met, then we're in normal operation/are the only one using the IAD.

File: lib/ListManager.js [hunk 4]
```diff
@@ -0,0 +1,781 @@
...
+            // reset block assignment doc
+            const {content} = x.blockAssignmentDoc;
+            content.rlSequence = x.item.rlSequence;
+            content.blockCount = BLOCK_COUNT;
+            content.blockSize = BLOCK_SIZE;
+            // TODO: optimize by using string from config or static var
+            const bs = new Bitstring({length: content.blockCount});
+            content.assignedBlocks = await bs.encodeBits();
+            content.assignedBlockCount = 0;
+            // TODO: what else?
```
[the remark this answers is not in the document]
Comment (dlongley) [reply lib/ListManager.js:287]: +1, I think I left them there not knowing what else might be needed. I'll remove them.

File: lib/http.js (hunk 2 above)
Comment (dlongley) [reply lib/http.js:127]: Yeah, but not tested.

**Pull request merged**